tools/windows-node-installer/test/e2e/azure_test.go

```diff
@@ -354,8 +363,29 @@ func testAnsiblePing(t *testing.T) {
 
 // testAzureFirewallRule asserts if the created instance has firewall rule that opens container logs port.
 func testAzureInstancesFirewallRule(t *testing.T) {
+	var err error
+	windowsVM, err = getAzureCreds()
+	require.NoErrorf(t, err, "failed to obtain credentials for %s", credentials.GetInstanceId())
+	testInstanceFirewallRule(t)
+}
+
+func getAzureCreds() (windowsVM types.WindowsVM, err error) {
+	w := &types.Windows{}
+	ipAddress, password, err := getIpPass(credentials.GetInstanceId())
```

Comment:
I think you should early if you got an error here

Reply:
will do it